draft-ietf-tls-tls13.md

```diff
@@ -2312,7 +2320,47 @@ the existing behavior for SignedCertificateTimestamps {{?RFC6962}},
 and is more easily extensible in the handshake state machine.
 
 
-## Server Parameters Messages
+### Hello Finished
+
+The "hello_finished" extension is used to carry a Finished.verify_data
+({{finished}}) in the ClientHello. The reason for this structure rather
+than sending the Finished as a separate message is for backward
+compatibility: the client can send a "hello_finished" extension
+to servers which may not support TLS 1.3 without causing them to
+fail the handshake. This allows for mixed server configurations which
+concurrently support TLS 1.3 (including PSK resumption but not 0-RTT)
+and TLS 1.2 without creating interoperability problems.
+
+The "hello_finished" extension MUST be sent if the client offers a
+"pre_shared_key" extension and MUST NOT be sent otherwise. The Hash
+used in the computation is that associated with the indicated PSK.
+Servers MUST NOT send this extension. This extension MUST be the
+last extension in the ClientHello (this facilitates implementation
+as described below). Servers MUST check that it is the last
+extension and otherwise fail the handshake with an "illegal_parameter"
+alert.
+
+The contents of the extension are simply a Finished value, with the
+Handshake Context being the ClientHello message with the verify_data
+field of the extension being set to a string of Hash.length zeroes.
+One way to implement this is:
+
+1. Marshall the entire ClientHello with zeroes in the verify_data
+   location.
+2. Compute the "hello_finished" extension
+3. Copy the verify_data into the last Hash.length bytes of the
+   ClientHello.
+
+This means that the "hello_finished" is computed over a semantically
+valid ClientHello message where the length of the message is consistent
+with the number of bytes hashed.
+
+The actual ClientHello, complete with the correct verify_data, is included
+in all other handshake hash computations.
+
+[[OPEN ISSUE: Should we instead hash the first length - Hash.length() bytes?]]
+
+## Server Parameters
```

Comment:
Nit: This seems to have lost the 'Messages' suffix which all the other sections have. Although "parameters messages" sounds kinda weird. "Server Parameter Messages" maybe??